docs/DifferentiableTypes.md

```diff
@@ -0,0 +1,289 @@
+# Differentiable types
+
+Richard Wei, Dan Zheng
+
+March 2019
+
+## Preface
+
+Mathematically speaking, only functions are "differentiable": only functions have derivatives and can be differentiated. In this document, the terminology "differentiable types" is used as a shorthand for "types that can be used as arguments and results of differentiable functions". This notion is important because not all types are "differentiable" in this sense. For example, types representing real numbers and vector spaces are "differentiable", but strings and integers are not.
```

Comment:
```suggestion
Speaking in terms of elementary calculus, only functions are "differentiable": only functions have derivatives and can be differentiated. In this document, the terminology "differentiable types" is used as a shorthand for "types that can be used as arguments and results of differentiable functions". This notion is important because not all types are "differentiable" in this sense. For example, types representing real numbers and vector spaces are "differentiable", but strings and integers are not.
```

Mathematically speaking, spaces can be differentiable as well.